tests/ui/inference/question-mark-type-infer.stderr

```diff
@@ -1,14 +1,35 @@
-error[E0282]: type annotations needed
+error[E0283]: type annotations needed
   --> $DIR/question-mark-type-infer.rs:10:21
    |
 LL |     l.iter().map(f).collect()?
    |                     ^^^^^^^ cannot infer type of the type parameter `B` declared on the method `collect`
    |
+   = note: multiple `impl`s satisfying `_: FromIterator<Result<i32, ()>>` found in the following crates: `alloc`, `core`, `hashbrown`, `std`:
+           - impl<'a, T> FromIterator<T> for Cow<'a, [T]>
+             where T: Clone;
+           - impl<A, E, V> FromIterator<Result<A, E>> for Result<V, E>
+             where V: FromIterator<A>;
+           - impl<I> FromIterator<I> for Box<[I]>;
+           - impl<T, S, A> FromIterator<T> for hashbrown::set::HashSet<T, S, A>
+             where T: Eq, T: Hash, S: BuildHasher, S: Default, A: Default, A: Allocator, A: Clone;
+           - impl<T, S> FromIterator<T> for HashSet<T, S>
+             where T: Eq, T: Hash, S: BuildHasher, S: Default;
+           - impl<T> FromIterator<T> for Arc<[T]>;
+           - impl<T> FromIterator<T> for BTreeSet<T>
+             where T: Ord;
+           - impl<T> FromIterator<T> for BinaryHeap<T>
+             where T: Ord;
+           - impl<T> FromIterator<T> for LinkedList<T>;
+           - impl<T> FromIterator<T> for Rc<[T]>;
+           - impl<T> FromIterator<T> for Vec<T>;
+           - impl<T> FromIterator<T> for VecDeque<T>;
```

Comment:
I expect this to be quite overwhelming to a user. Why is it beneficial to show an exhaustive list of impls?

Reply:
My next step would be to special case the `From`, `Into`, `FromIterator` and `IntoIterator` impls to extract the right types for the structured suggestions and try to come up with some heuristic for which types you most likely intended. I can make the list shorter again and maybe display a handful and "and N more" like we do elsewhere.

Reply:
done